### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by Alvaro8gb**
I convert the notebook to .py

### CODE REVIEW

File: Assignment4/Alvaro8gb-23A257/Task07.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+print("RDFlib:")
+for s, p, o in g.triples((None, RDFS.subClassOf, NS.LivingThing)):
+    print(s)
+
+print("SPARQL:")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE {
+    ?Subject rdfs:subClassOf ns:LivingThing.
+  }
+  ''',
```
Comment (anaigmo) [inline Assignment4/Alvaro8gb-23A257/Task07.py:50]: and subclasses of subclasses?

File: Assignment4/Alvaro8gb-23A257/Task07.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+# 
+
+# In[6]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFLIB")
+for c in [NS.Person, NS.Animal]:
+    for s, p, o in g.triples((None, RDF.type, NS.Person)):
```
Comment (anaigmo) [inline Assignment4/Alvaro8gb-23A257/Task07.py:97]: only type person? and animal (above)?

Comment (Alvaro8gb) [reply Assignment4/Alvaro8gb-23A257/Task07.py:97]: thanks for the comment I will do a pull request with a better solution

**Comment by Alvaro8gb** [comment]
bugs fixed

File: Assignment4/Alvaro8gb-23A257/Task07.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[4]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFlib:")
+for s, p, o in g.triples((None, RDFS.subClassOf, NS.LivingThing)):
```
Comment (dgarijo) [inline Assignment4/Alvaro8gb-23A257/Task07.py:42]: missing sublcasses of subclasses of LivingThing

Comment (dgarijo) [reply Assignment4/Alvaro8gb-23A257/Task07.py:42]: hint: make a recursive function

File: Assignment4/Alvaro8gb-23A257/Task07.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+# In[5]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFlib:")
+for s, p, o in g.triples((None, RDF.type, NS.Person)):
+    print(s)
+
+for s1, p1, o1 in g.triples((None, RDFS.subClassOf, NS.Person)):
```
Comment (dgarijo) [inline Assignment4/Alvaro8gb-23A257/Task07.py:69]: same problem as above

**Review by dgarijo — CHANGES_REQUESTED** [review]
You almost have it!

**Comment by Alvaro8gb** [comment]
I think now is fine

**Review by dgarijo — APPROVED** [review]

**Pull request merged**